include/deal.II/base/tensor.h

```diff
@@ -205,6 +205,16 @@ class Tensor<0,dim,Number>
   template <typename OtherNumber>
   Tensor<0,dim,Number> &operator = (const Tensor<0,dim,OtherNumber> &rhs);
 
+  /**
+   * Assignment from tensors with same underlying scalar type.
+   */
+  Tensor<0,dim,Number> &operator = (const Tensor<0,dim,Number> &rhs);
```

Comment:
Would it be possible to use `= default` here?

Reply:
I am surprised that we need this in addition to the more general template above. This compiler is weird.

Reply:
I think that the copy operators we do declare prevent the compiler from providing the default operator.

Reply:
I second the suggestion for using `=default` here.

Reply:
May you please also remove the `<0,dim,Number>` where on the input and return type both here and below. This would then align with #5848.

Reply:
The default one doesn't work.

```
In file included from /home/qiaol/devel/dealii_git/source/fe/fe_values.cc(4587):
/home/qiaol/devel/dealii_git/include/deal.II/base/tensor.h(211): error: the parameter of this defaulted assignment operator cannot be const because a base or member copy assignment parameter is non-const
    Tensor<0,dim,Number> &operator = (const Tensor<0,dim,Number> &rhs) = default;
                          ^
In file included from /home/qiaol/devel/dealii_git/source/fe/fe_values.cc(4587):
          detected during:
In file included from /home/qiaol/devel/dealii_git/source/fe/fe_values.cc(4587):
            instantiation of class "dealii::Tensor<0, dim, Number> [with dim=1, Number=Sacado::Rad::ADvar<double>]" at line 403
In file included from /home/qiaol/devel/dealii_git/source/fe/fe_values.cc(4587):
            instantiation of class "dealii::Tensor<rank_, dim, Number> [with rank_=1, dim=1, Number=Sacado::Rad::ADvar<double>]" at line 389 of "/home/qiaol/devel/dealii_git/include/deal.II/base/template_constraints.h"
In file included from /home/qiaol/devel/dealii_git/source/fe/fe_values.cc(4587):
            instantiation of class "dealii::internal::ProductTypeImpl<T, U> [with T=Sacado::Rad::ADvar<double>, U=dealii::Tensor<1, 1, double>]" at line 448 of "/home/qiaol/devel/dealii_git/include/deal.II/base/template_constraints.h"
In file included from /home/qiaol/devel/dealii_git/source/fe/fe_values.cc(4587):
            instantiation of class "dealii::ProductType<T, U> [with T=Sacado::Rad::ADvar<double>, U=dealii::Tensor<1, 1, double>]" at line 187 of "/home/qiaol/devel/dealii_git/include/deal.II/fe/fe_values.h"
In file included from /home/qiaol/devel/dealii_git/source/fe/fe_values.cc(4587):
            instantiation of class "dealii::FEValuesViews::Scalar<dim, spacedim>::OutputType<Number> [with dim=1, spacedim=1, Number=Sacado::Rad::ADvar<double>]" at line 24629 of "/home/qiaol/package_install/deal.II-git/build_devel_icc/source/fe/fe_values.inst"

```